User-Data Store Migration/README.md

```diff
@@ -0,0 +1,85 @@
+
+# The migration process
+
+Linkurious Enterprise is designed to create and maintain (through the updates) it's data structure over the supported databases.
+
+The safest migration approach is to let the system handle the whole task of defining the db schema and then perform manual tasks only to reimport the data.
+
+The tool is designed to help a database administrator to generate the correct script to import data staring from a dump. In particular the script will:
+1. remove all the `DDL` instructions;
+2. change all the `insert` instructions to explicitly add the field name (to ensure the correct mapping of data regardless possible difference in the order);
+3. `Truncate` all the table's contents before data insert (it's needed to delete any default data generated by the system during the schema creation);
+4. correct any known compatibility issue between the output syntax of sqlite3 dump and the syntax used by the supported databases;
+5. list in output all the `select` instructions needed to access all the migrated data, it could be useful for running any query to verify the correctness of the migration.
+
+
+The full list of steps would be:
+1. Create a text dump of the current Sqlite file with the sqlite3 tool:
+   ```shell
+   sqlite3 ./linkurious/data/server/database.sqlite .dump > export.sql
+   ```
+2. Put the `export.sql` file in the same directory of the `dump-converter.py` python script
+3. Run the program to generate a new SQL import script (example for MySQL):
+   ```shell
+   python3 dump-converter.py --dialect mysql export.sql > select-queries.sql
+   ```
+4. The program will generate two files:
+   - export-parsed.sql : the new SQL import file to be used instead of the initial one
+   - select-queries.sql : a script containing the `select` queries for all the tables found in the initial script. It can be useful for debugging / crosscheck purposes.
+5. Setup your external database server and create an empty database
+6. Configure Linkurious Enterprise to point the new Database (refer to the correct version of the [documentation](https://doc.linkurio.us) in case of doubts)
+7. Start Linkurious Enterprise
+8. As soon as it starts properly (you can see the application, regardless the connection error to the graph database), stop it.
+9. Run the new script in your system to import data (example for MySQL):
+   ```shell
+   mysql -u MY_MYSQL_USER -p -h 127.0.0.1 linkurious < export-parsed.sql
+   ```
+10. After the import script is executed successfully you can start again Linkurious Enterprise and accessing all the previous data
+
+
+# How to use / customize the tool
+
+The script is develop to work without any change based on the current known issues at the time of the development. Few parameters are possible to configure the destination database or the input / output file, below the extract of the help.
```

Comment:
"develop" ~> "developed"
"Few parameters are possible to configure..." ~> "With few parameters it's possible to configure..."